File: Lab 7/Taktau/app/src/main/java/com/example/taktau/MainActivity.java
package com.example.taktau;


import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    SensorManager salesman;
    ListView lv;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        lv = (ListView)findViewById(R.id.listview);

        salesman = (SensorManager)getSystemService(SENSOR_SERVICE);
        List<Sensor> sensorList = salesman.getSensorList(Sensor.TYPE_ALL);

        lv.setAdapter(new ArrayAdapter<Sensor>(this, android.R.layout.simple_list_item_1, sensorList));
    }
}

File: Lab 7/Task1B/app/src/main/java/com/example/task1b/MainActivity.kt
package com.example.task1b

import android.annotation.SuppressLint
import android.hardware.Sensor
import android.hardware.SensorManager
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var salesman: SensorManager
    private lateinit var lv: ListView

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lv = findViewById(R.id.listview)

        salesman = getSystemService(SENSOR_SERVICE) as SensorManager
        val sensorList: List<Sensor> = salesman.getSensorList(Sensor.TYPE_ALL)

        // Use 'this' as the context in the ArrayAdapter constructor
        lv.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, sensorList)
    }
}
